### PR TITLE
ajd chapter 5

### PR DESCRIPTION
**Pull request by alexdeas**
ajd chapter 5

### CODE REVIEW

File: chapter05/ajd/exercise2.pl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+# Get the file read
+my %total_bytes;
+
+while ( <> ) {
+  if ( /^#/) { next };
+  my ( $source, $destination, $bytes ) = split;
+  $total_bytes{$source}{$destination} += $bytes;
+}
```
Comment (utopia-2) [inline chapter05/ajd/exercise2.pl:17]: :+1: for not doing the extra sum in the file read loop.

File: chapter05/ajd/exercise2.pl [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+# Sorting and printing
+foreach my $host ( sort { by_host_data( $a, $b, \%total_bytes ) } keys %total_bytes ) {
```
Comment (utopia-2) [inline chapter05/ajd/exercise2.pl:20]: This'll have to total the bytes for each host an unnecessary number of times to sort the hashes. Probably worth pre-computing the totals and sorting on that.

File: chapter05/ajd/exercise2.pl [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  foreach my $dest ( sort { by_dest_data( $a, $b, $total_bytes{$host} ) } keys $total_bytes{$host} ) {
```
Comment (utopia-2) [inline chapter05/ajd/exercise2.pl:22]: The body of the sort would probably be clearer _without_ the function call in this case.

Comment (alexdeas) [inline chapter05/ajd/exercise2.pl:22]: True, I think I wanted to keep it like the other one when I wrote it.

File: chapter05/ajd/exercise2.pl (hunk 2 above)
Comment (alexdeas) [inline chapter05/ajd/exercise2.pl:20]: Yeah, there's a bit about that in a later chapter (spoilers).

File: chapter05/ajd/exercise1.pl [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+my %hash = (
+  # some stuff
+);
+
+# Or
+
+my $hashref = {
+  # some stuff
+};
+
```
Comment (SimonGross) [inline chapter05/ajd/exercise1.pl:18]: :+1: :+1:

Double thumbs for demonstrating both correct versions.

**Pull request merged**